ostia-operator/test/integration.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+OPENSHIFT_PUBLIC_HOSTNAME=${1}
+
+# Check if namespace exists
+# Arg: Project name
+project_exists () {
+  oc get project $1 &> /dev/null
+}
+
+# Delete namespace if it exists
+# Arg: Project name
+delete_project_and_wait () {
+  if project_exists $1; then
+    echo "Cleaning up project" ${1}
+    oc delete project ${1} --now=true
+    while project_exists ${1}
+    do
+        sleep 3
+    done
+   fi
+}
+
+# Give Pods two minutes to hit ready status before error
+# Expects two args: Key and value filter for Pod metadata labels
+wait_for_pod_ready () {
+ sleep 3 # Wait for Pod to be created before checking its readiness
+ retries="12"
+ while [[ "retries" -gt "0" ]]
+ do
+    if "$(oc get pod -o jsonpath='{.items[?(@.metadata.labels.'${1}'=="'${2}'")].status.containerStatuses[0].ready}')" == "true"
+    then
+        return
+    else
+        let "retries--"
+        sleep 10
+    fi
+ done
+ echo "Pod was not ready in time"
+ exit 1
+}
+
+# Prints to stdout the number of each unique status codes
+# Expects two args: Endpoint to make HTTP request against, number of requests to make
+do_http_get() {
+for i in $(seq 1 ${2}); do
+ curl -k -s -o /dev/null -w "%{http_code}\n" ${1};
+done | uniq -c
+
+}
+
+echo "Testing Operator Deployment"
```

Comment:
It would be nice if we can do something like: 

`run_tests ./integration_tests`  and it just goes through some small scripts: 

```
01_deploy_operator.sh
02_simple_endpoint.sh
03_check_endpoint_reconciliation.sh 
04_ .....
```
Importing a common library, for retry/wait/cleanup etc.. 

wdyt?

Reply:
As we have been discussing, perhaps making those tests self-contained, can be a good idea: 

Each individual test, should: 

1. Create a  project with a random name
2. Deploy the operator with the "NAMESPACE" env var pointing to itself
3. Do the test.
4. Show results and teardown. 

This way we can parallelize the whole thing without too much effort.